client/container_list.go

```diff
@@ -35,7 +37,14 @@ func (cli *Client) ContainerList(ctx context.Context, options types.ContainerLis
 	}
 
 	if options.Filter.Len() > 0 {
-		filterJSON, err := filters.ToParam(options.Filter)
+		filterJSON := ""
+		err := errors.New("")
+		// for daemons older than v1.10, filter must be of the form map[string][]string
+		if cli.version != "" && version.Version(cli.version).LessThan(version.Version("1.10")) {
```

Comment:
`1.10` -> CLI should deal with API version, not Engine version. 
https://docs.docker.com/engine/reference/api/docker_remote_api/